cpp/src/arrow/flight/internal.cc

```diff
@@ -45,12 +46,64 @@ namespace flight {
 namespace internal {
 
 const char* kGrpcAuthHeader = "auth-token-bin";
+const char* kGrpcStatusCodeHeader = "x-arrow-status";
+const char* kGrpcStatusMessageHeader = "x-arrow-status-message-bin";
+const char* kGrpcStatusDetailHeader = "x-arrow-status-detail-bin";
+
+/// Try to extract a status from gRPC trailers. If not found, return
+/// Status::OK.
+Status FromGrpcContext(const grpc::ClientContext& ctx) {
+  const std::multimap<grpc::string_ref, grpc::string_ref> trailers =
```

Comment:
`const&`, no? It's wasteful to make a copy.

Reply:
Made this const&, yes.